src/char_mapping.rs

```diff
@@ -187,7 +285,44 @@ impl Encoding4 {
 
 impl Encoding6 {
     /// Return the mapping.
-    pub fn mapping(&self) -> HashMap<u16, u16> {
-        unimplemented!();
+    pub fn mapping(&self) -> HashMap<u16, GlyphID> {
+        let mut map = HashMap::new();
```

Comment:
I think the names of the variables here and in the tests are left. Please rename `map` to `mapping`, `buf` to `buffer`, and `vsr` to `record`.